third_party/terraform/resources/resource_compute_project_metadata_item.go

```diff
@@ -56,7 +56,7 @@ func resourceComputeProjectMetadataItemCreate(d *schema.ResourceData, meta inter
 	key := d.Get("key").(string)
 	val := d.Get("value").(string)
 
-	err = updateComputeCommonInstanceMetadata(config, projectID, key, &val, int(d.Timeout(schema.TimeoutCreate).Minutes()))
+	err = updateComputeCommonInstanceMetadata(config, projectID, key, &val, int(d.Timeout(schema.TimeoutCreate).Minutes()), true)
```

Comment:
nit: You can "name" the values provided to the function by creating a type instead of using raw bools.

```go
type metadataPresentBehavior bool

const (
	failIfPresent metadataPresentBehavior = true
	expectPresent metadataPresentBehavior = false
)
```

The behaviour of the function ends up being a little easier to read from the callsite, instead of needing to check the param names. 

`updateComputeCommonInstanceMetadata(config, projectID, key, &val, int(d.Timeout(schema.TimeoutCreate).Minutes()), failIfPresent)`